diff --git a/node_modules/@astrojs/starlight/components/TableOfContents/MobileTableOfContents.astro b/node_modules/@astrojs/starlight/components/TableOfContents/MobileTableOfContents.astro
index 142f15c..cea66ca 100644
--- a/node_modules/@astrojs/starlight/components/TableOfContents/MobileTableOfContents.astro
+++ b/node_modules/@astrojs/starlight/components/TableOfContents/MobileTableOfContents.astro
@@ -35,6 +35,7 @@ const t = useTranslations(locale);
 <style>
   nav {
     position: fixed;
+    z-index: 1;
     top: calc(var(--sl-nav-height) - 1px);
     inset-inline: 0;
     border-top: 1px solid var(--sl-color-gray-5);
